frappe/public/js/frappe/form/grid.js

```diff
@@ -489,7 +494,12 @@ export default class Grid {
 				if (!this.df.data) {
 					this.df.data = this.get_data() || [];
 				}
-				this.df.data.push({idx: this.df.data.length+1, __islocal: true});
+				var d = { idx: this.df.data.length + 1, __islocal: true, doctype: this.doctype, parentfield: this.df.fieldname, parenttype: frappe.web_form.doctype }
+				this.df.data.push(d);
+				if(!frappe.web_form.doc.hasOwnProperty(this.df.fieldname)) {
```

Comment:
Let's just use the `in` operator to do this check:

```suggestion
				if (!(this.df.fieldname in frappe.web_form.doc)) {
```